VideoLocker/src/main/java/org/edx/mobile/util/ResourceUtil.java

```diff
@@ -1,64 +1,59 @@
 package org.edx.mobile.util;
 
 import android.content.Context;
+import android.support.annotation.PluralsRes;
+import android.support.annotation.StringRes;
 
 import com.squareup.phrase.Phrase;
 
-import org.edx.mobile.R;
 import org.edx.mobile.base.MainApplication;
 
+import java.util.HashMap;
+import java.util.Set;
+
 
 public class ResourceUtil {
     public static final String QuantityHolder = "quantity";
 
-
-    public static String getResourceString(String name) {
-        Context context = MainApplication.instance().getApplicationContext();
-        return getResourceString(name, context);
-    }
-
-    /**
-     * get the string resources dynamically
-     * @param name
-     * @param context
-     * @return
-     */
-    public static String getResourceString(String name, Context context) {
-        int nameResourceID = context.getResources().getIdentifier(name, "string", context.getApplicationInfo().packageName);
-        if (nameResourceID == 0) {
-            throw new IllegalArgumentException("No resource string found with name " + name);
-        } else {
-            return context.getString(nameResourceID);
-        }
-    }
-
-    public static String getResourceString(int resourceId){
+    public static String getResourceString(@StringRes int resourceId) {
         Context context = MainApplication.instance().getApplicationContext();
         return context.getString(resourceId);
     }
 
-    public static CharSequence getFormattedString(int resourceId, String key, String value){
-        if ( value == null )
+    public static CharSequence getFormattedString(@StringRes int resourceId, String key, String value) {
+        if (value == null)
             value = "";
         return Phrase.from(ResourceUtil.getResourceString(resourceId))
-                .put(key, value) .format();
+                .put(key, value).format();
     }
 
+    public static CharSequence getFormattedString(@StringRes int resourceId, HashMap<String, CharSequence> keyValMap) {
```

Comment:
That's a whole lot more complicated than a Map. A builder pattern would be more appropriate, though this isn't a bottle-neck that we need to be optimizing right now.